src/test/kotlin/org/jetbrains/intellij/test/localIdeTestUtils.kt

```diff
@@ -0,0 +1,56 @@
+package org.jetbrains.intellij.test
+
+import org.jetbrains.intellij.IntelliJPluginConstants
+import java.io.File
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.zip.ZipFile
+
+/**
+ * Downloads and extracts IDE for the tests using local IDE installation. IDEs are downloaded from
+ * [IntelliJPluginConstants.DEFAULT_INTELLIJ_REPOSITORY].
+ *
+ * @param localIdesPath directory to store local IDE
+ * @param releasePath IDE path relative to [IntelliJPluginConstants.DEFAULT_INTELLIJ_REPOSITORY]/releases, e.g.
+ * `"com/jetbrains/intellij/idea/ideaIC/2021.2.4/ideaIC-2021.2.4.zip"`
+ */
+fun createLocalIdeIfNotExists(localIdesPath: Path, releasePath: String): String {
+    val fileName = releasePath.substringAfterLast('/')
+    val localIdeZipPath = localIdesPath.resolve(fileName)
+    val localIdeDirPathString = localIdeZipPath.toString().removeSuffix(".zip")
+    if (Files.exists(Path.of(localIdeDirPathString))) {
+        return localIdeDirPathString
+    }
+    if (!Files.exists(localIdesPath)) {
+        Files.createDirectories(localIdesPath)
+    }
+    URL("${IntelliJPluginConstants.DEFAULT_INTELLIJ_REPOSITORY}/releases/$releasePath")
+        .openStream().use { Files.copy(it, localIdeZipPath) }
+    localIdeZipPath.toFile().unzip()
+    Files.delete(localIdeZipPath)
+    return localIdeDirPathString
+}
+
+private fun File.unzip() {
+    val unzipDirPath = this.path.removeSuffix(".zip")
+    ZipFile(this).use { zip ->
```

Comment:
I don't see anything easily usable out of Gradle context. I considered using Zip4j but wanted to avoid adding a dependency. I'm not sure what is less evil here (dependency vs 15 lines of code).
Let me know if I should improve it.

Reply:
Agreed to skip it for now.